pygments/lexers/r.py

```diff
@@ -89,7 +89,7 @@ class SLexer(RegexLexer):
             (r'\[{1,2}|\]{1,2}|\(|\)|;|,', Punctuation),
         ],
         'keywords': [
-            (r'(if|else|for|while|repeat|in|next|break|return|switch|function)'
+            (r'(if|for|while|return|switch|function)'
```

Comment:
I am confused. This rule is apparently broken, since `include('valid_name')` comes before `include('keywords')` in the `'statements'` state. Do you really want to highlight these as functions and not as keywords? How about simply moving `include('valid_name')` after `include('keywords')`? Could you explain the reason why `else`, `repeat` etc. are different from `if`, `for`, etc.?

Reply:
I've been looking at the R specification and it does claim that things like "break" are indeed functions in https://cran.r-project.org/doc/manuals/r-release/R-lang.html#Flow-control-elements, but, in other sections they're considered keywords:

* https://cran.r-project.org/doc/manuals/r-release/R-lang.html#Looping
* https://cran.r-project.org/doc/manuals/r-release/R-lang.html#Reserved-words

I'm inclined to not change them to `Function` because they seem quite different from actual functions like `mean` etc.

Reply:
>>> I'm inclined to not change them to Function because they seem quite different from actual functions like mean etc.
> library(highr)   ## in the R console
> highr:::cmd_latex
                         cmd1 cmd2
COMMENT              \\hlcom{    }
FUNCTION             \\hlkwa{    }
IF                   \\hlkwa{    }
ELSE                 \\hlkwa{    }
WHILE                \\hlkwa{    }
FOR                  \\hlkwa{    }
IN                   \\hlkwa{    }
BREAK                \\hlkwa{    }
REPEAT               \\hlkwa{    }
NEXT                 \\hlkwa{    }
NULL_CONST           \\hlkwa{    }
LEFT_ASSIGN          \\hlkwb{    }

The same highlighting LaTeX command 
\hlkwa{ ...  }
is used for FUNCTION as for ELSE, BREAK, NEXT, IN, REPEAT 

But in SLexer ("r.py") actually highlight only function names (if followed by '(' evidently), not the keyword 'break', 'else' etc.
(except when they are followed by '(' - which is correct, but not usual; we write "for i in 1:5", not "for i in() 1:5")

--vladbazon